base/sort.jl

```diff
@@ -629,7 +709,12 @@ function selectperm!(ix::AbstractVector{<:Integer}, v::AbstractVector,
 
     # do partial quicksort
     sort!(ix, PartialQuickSort(k), Perm(ord(lt, by, rev, order), v))
-    return ix[k]
+
+    if k isa Integer
```

Comment:
Likewise here, perhaps using dispatch for this purpose would be better than branching?